GitCommands/Patches/PatchProcessor.cs

```diff
@@ -71,71 +89,57 @@ public static IEnumerable<Patch> CreatePatchesFromString(string patchText, Lazy<
                 return null;
             }
 
-            string header = lines[lineIndex];
-
-            Match headerMatch = PatchHeaderRegex().Match(header);
-
+            string rawHeader = lines[lineIndex];
+            Match contentMatch = StripWrappingEscapesSequenceRegex().Match(rawHeader);
+            string header = contentMatch.Groups["line"].Value;
+#if DEBUG
+            DebugHelpers.Assert(!CheckAnyEscapeSequenceRegex().IsMatch(header), "Git unexpectedly emits escape sequences in first header other than at start/end. line {i}:{rawHeader}");
+#endif
```

Comment:
`DebugHelpers.Assert` is already debug-only as `Debug.WriteLineIf`, too, which could be an alternative.
```suggestion
            DebugHelpers.Assert(!CheckAnyEscapeSequenceRegex().IsMatch(header), "Git unexpectedly emits escape sequences in first header other than at start/end. line {i}:{rawHeader}");
```

Reply:
But `CheckAnyEscapeSequenceRegex()` is in DEBUG so this change will require that the Regex is included in Release builds (at least as a dummy returning false). I suggest to keep this as is.

Reply:
> But `CheckAnyEscapeSequenceRegex()` is in DEBUG so this change will require that the Regex is included in Release builds (at least as a dummy returning false). I suggest to keep this as is.

OK, I see now. It seems to be the week of unexpected limitations of C#.